leftwm-core/src/handlers/command_handler.rs

```diff
@@ -567,6 +568,28 @@ fn focus_window_change(
     Some(handle_focus(state, handle))
 }
 
+fn focus_window_top(state: &mut State, toggle: bool) -> Option<bool> {
+    let tag = state.focus_manager.tag(0)?;
+    let cur = state.focus_manager.window(&state.windows).map(|w| w.handle);
+    let prev = state.focus_manager.tags_last_window.get(&tag).copied();
+    let next = state
+        .windows
+        .iter()
+        .find(|x| x.tags.contains(&tag) && !x.is_unmanaged())
+        .map(|w| w.handle);
+
+    match (next, cur, prev) {
+        (Some(next), Some(cur), Some(prev)) if next == cur && toggle => {
+            Some(handle_focus(state, prev))
+        }
+        (Some(next), Some(cur), _) if next != cur => {
+            state.focus_manager.tags_last_window.insert(tag, cur);
```

Comment:
Shouldn't this be done by `handle_focus`?

Reply:
Yes this would be handled by focus_window which is called by handle_focus.

Reply:
It does not seems to be the case, `focus_window` only add it to `window_history` and not `tags_last_windows`, and the test fails If I remove this

Reply:
Oh yeah sorry misread it. At some point we should probably add it there, as currently we save of only when changing tag or reloading.